domain/src/main/java/org/oppia/domain/topic/TopicController.kt

```diff
@@ -186,6 +220,170 @@ class TopicController @Inject constructor(
     )
   }
 
+  fun getCompletedStoryList(profileId: ProfileId): LiveData<AsyncResult<CompletedStoryList>> {
+    return dataProviders.convertToLiveData(retrieveCompletedStoryListDataProvider(profileId))
+  }
+
+  fun getOngoingTopicList(profileId: ProfileId): LiveData<AsyncResult<OngoingTopicList>> {
+    return dataProviders.convertToLiveData(retrieveOngoingTopicListDataProvider(profileId))
+  }
+
+  private fun retrieveCompletedStoryListDataProvider(profileId: ProfileId): DataProvider<CompletedStoryList> {
+    return dataProviders.transformAsync(
+      TRANSFORMED_GET_COMPLETED_STORIES_PROVIDER_ID,
+      storyProgressController.retrieveCacheStore(profileId)
+    ) {
+      val completedStoryListBuilder = CompletedStoryList.newBuilder()
+      it.topicProgressMap.keys.forEach { topicId ->
```

Comment:
Suggest naming the variable rather than relying on ``it`` since there are several levels of lambdas here.

Reply:
Done here and everywhere else.